### DISCUSSION TITLE
Transparent ring offset color

### DISCUSSION THREAD

**Discussion by jellebouwman** [tailwindlabs/tailwindcss#5257] [Help]
Hi! 👋 

## Goal
I've been using the [ring utility](https://tailwindcss.com/docs/ring-width) classes for some of my focusable UI elements. I'm currently running into an issue creating transparent `ring-offset`s on gradient backgrounds, and I'm wondering if I can solve this with Tailwind.

I'm trying to understand what the appropriate usecase is for `ring-color-transparent`.

## Context
- I have a modal with a gradient background.
- On this modal, I have a couple of buttons I'd like to interact with.
- When I focus these buttons, I'd like to show a focus ring, with a bit of offset.

We can style the `ring-offset` in a couple of ways, including color. The docs state that Tailwind uses `white` as a default, and if you'd like to match the offset with the background color you use, the docs suggest that you match it.

```
<div class="bg-cyan-300>
  <button class="
      /* ring offset styles */
      ring-offset-cyan-300 ring-offset-2
      /* ring styles */
      tw-ring-2 tw-ring-red-300
      ... /*other styles */
    ">Click me</button>
</button>
```

Because I'm using a `linear-gradient` as a background, I don't have an exact color I can use for the `ring-offset` utility. When looking at the docs, I hoped `ring-color-transparent` could help me out here, but it this utility variant is working differently than I expected - even without a `linear-gradient` background.

### Playground
I've created a sandbox environment where I reproduced the behaviour I'm running into. You can find it [here](https://play.tailwindcss.com/1Lbl4wVMwz).

**Comment by d8vjork** [comment]
This is the case I've when using dark colors as for the background on my websites (using the dark mode for example)

Can't manage to change the color unless there's some manual workaround on this...

**Comment by simonswiss** [comment]
Hey @jellebouwman!

Here's why it behaves differently than you expected:

The `ring` and `ring-offset` are achieved with overlapping box-shadows. When you add a `ring-offset`, it doesn't actually move the "halo" (or ring) away from the element – it just creates another box-shadow over the top of that ring, and expands the ring box-shadow itself to give the impression that it's offset 😅

So, basically, the `ring` color is also there under the `ring-offset` color.

Does it make sense?

So, when you apply a `ring-offset-transparent` color to the `ring-offset`, what you will see through is... the `ring` box-shadow color that was underneath!

I have just put together a quick concept here, and this might very well be a bad idea, but it does the job:

1) wrap your button in a div with some padding, and a border color (I guess `ring` would also work).
2) set the opacity of that border to `0`.
3) since you want this border (fake focus ring) to show when the button is focused, use `focus-within:border-opacity-100` to reveal that border when the button is in focus

🎉

![image](https://user-images.githubusercontent.com/485747/133546292-ada2af4a-2a77-4f11-bd98-5ce8ddeef445.png)

Here I had to apply a greater border-radius on the wrapping element to make it look good, which is not always going to work great, but looks decent in that case!

Here's the Play I came up with when reading this question:

https://play.tailwindcss.com/EN0CYCZmQ4

Hope it helps!

**Reply by jellebouwman** [reply]
Thanks @simonswiss - this looks like what I want to accomplish! Appreciate the elaborate answer, will write back if I gave it a shot!

**Comment by Chong985** [comment]
I appreciate what you did.

[liteblue](https://www.liteblue.life/)

**Comment by vhscom** [comment]
Opened a bug: https://github.com/tailwindlabs/tailwindcss/issues/8128
